Sources/Build/BuildPlan.swift

```diff
@@ -312,7 +330,13 @@ public final class ClangTargetBuildDescription {
             else if let generatedModuleMapType = clangTarget.moduleMapType.generatedModuleMapType {
```

Comment:
Why does `clangTarget` need to change? Aren't we only concerned with `MixedTarget`?

Reply:
The `MixedTarget` just wraps a `ClangTarget` + `SwiftTarget` so the path forward with the least code changes was to make the `ClangTarget` generate the module map bits for a MixedTarget if it was told to (via the `generateSwiftHeaderInModuleMap` param).